core/modules/qproc/QuerySession.cc

```diff
@@ -404,31 +404,31 @@ std::ostream& operator<<(std::ostream& out, QuerySession const& querySession) {
 }
 
 
-ChunkQuerySpec QuerySession::buildChunkQuerySpec(query::QueryTemplate::Vect const& queryTemplates,
+ChunkQuerySpec::Ptr QuerySession::buildChunkQuerySpec(query::QueryTemplate::Vect const& queryTemplates,
                                                  ChunkSpec const& chunkSpec) const {
-    ChunkQuerySpec cQSpec;
-    cQSpec.db = _context->dominantDb;
-    cQSpec.scanInfo = _context->scanInfo;
-    cQSpec.scanInteractive = _scanInteractive;
-    cQSpec.chunkId = chunkSpec.chunkId;
+    ChunkQuerySpec::Ptr cQSpec = std::make_shared<ChunkQuerySpec>();
+    cQSpec->db = _context->dominantDb;
+    cQSpec->scanInfo = _context->scanInfo;
+    cQSpec->scanInteractive = _scanInteractive;
+    cQSpec->chunkId = chunkSpec.chunkId;
```

Comment:
I gave it a constructor, but the ChunkQuerySpec class really needs more work than that.